### PR TITLE
Revert delegation of configuration constants to autoloader

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

It is not possible to make the autoloading logic compatible with PHPUnit process isolation.

Instead of only reverting #17424, I moved the logic into a dedicated class.

### CODE REVIEW

**Comment by trasher** [comment]
Maybe you forgot to declare and set the property used in your last commit ;)

File: src/Glpi/Application/ConfigurationConstants.php [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Application;
+
+final class ConfigurationConstants
+{
```
Comment (trasher) [inline src/Glpi/Application/ConfigurationConstants.php:39]: private string $root_dir;

File: src/Glpi/Application/ConfigurationConstants.php [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    public function __construct(private string $root_dir)
+    {
```
Comment (trasher) [inline src/Glpi/Application/ConfigurationConstants.php:41]:
```suggestion
    {
        $this->root_dir = $root_dir;
```

**Comment by cedric-anne** [comment]
> Maybe you forgot to declare and set the property used in your last commit ;)

No, see https://stitcher.io/blog/constructor-promotion-in-php-8

**Review by trasher — APPROVED** [review]

**Pull request merged**